octodns/provider/base.py

```diff
@@ -30,6 +32,24 @@ def __init__(self, id, apply_disabled=False,
         self.delete_pcent_threshold = delete_pcent_threshold
         self.strict_supports = strict_supports
 
+    def _check_dynamic_for_private_ips(self, desired, record):
+        if record._type not in ('A', 'AAAA'):
+            # won't necessarily contain IPs
+            return
+        for name, pool in record.dynamic.pools.items():
+            for i, value in enumerate(pool.data['values']):
+                ip = ip_address(value['value'])
+                status = value['status']
+                if not ip.is_global and status != 'up':
+                    msg = f'non-routable dynamic record {record.fqdn} value ' \
+                        f'{ip} requires status `up` and has `{status}`'
+                    fallback = 'setting status to `up`'
+                    self.supports_warn_or_except(msg, fallback)
+                    record = record.copy()
+                    pool = record.dynamic.pools[name]
+                    pool.data['values'][i]['status'] = 'up'
```

Comment:
Azure DNS does not support `status: up` (supports `obey` and `down`), so this will mess up the apply phase. Since the base `process_desired_zone` is called after the provider's override, AzureProvider won't be able to fix it.